src/build.rs

```diff
@@ -259,17 +259,64 @@ impl BuildQueue {
             cmd.env("CARGO_TARGET_DIR", &Path::new("target").join("rls"));
             cmd.current_dir(build_dir);
 
-            let mut new_cmd_line = match cmd.output() {
+            let new_cmd_line_args = match cmd.output() {
                 Ok(output) => {
                     let out = String::from_utf8(output.stderr).unwrap();
                     // println!("output: `{}`", out);
                     let exit_str = "     Running `";
+
                     match out.rfind(exit_str) {
                         Some(i) => {
                             let remaining = &out[i + exit_str.len() ..];
                             let end = remaining.find('`').unwrap();
-                            remaining[..end].to_owned()
+                            let remaining = &remaining[..end];
+
+                            let mut result = vec![];
+
+                            let mut in_quoted_arg = false;
+                            let mut current_arg = String::with_capacity(64);
+
+                            for c in remaining.chars() {
+                                match (in_quoted_arg, c) {
+                                    (false, '"') => {
+                                        // Opening quote of quoted arg
+                                    },
```

Comment:
style nits: can you put the comments outside the match arm, then you can put the empty block on one line. I'd also prefer not using blocks and commas for the arm bodies.